providers/yttcb/clusterbootstrap.yaml

```diff
@@ -61,9 +91,26 @@
 #@ end
 
 #@ def should_create_clusterbootstrap():
-#@   return data.values.CNI != "antrea" or antrea_configs_exist() or vspherecsi_configs_exist() or vspherecpi_configs_exist() or awsebscsi_configs_exist() or azurefilecsi_configs_exist() or storage_class_configs_exist()
+#@   return data.values.CNI != "antrea" or antrea_configs_exist() or vspherecsi_configs_exist() or vspherecpi_configs_exist() or awsebscsi_configs_exist() or azurefilecsi_configs_exist() or storage_class_configs_exist() or pinniped_configs_exist()
 #@ end
 
+#@ if pinniped_configs_exist():
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: #@ "{}-pinniped-addon".format(data.values.CLUSTER_NAME)
+  namespace: #@ data.values.NAMESPACE
+  labels:
+    tkg.tanzu.vmware.com/addon-name: pinniped
+    tkg.tanzu.vmware.com/cluster-name: #@ data.values.CLUSTER_NAME
+    clusterctl.cluster.x-k8s.io/move: ""
+  annotations:
+    tkg.tanzu.vmware.com/addon-type: "authentication/pinniped"
+type: tkg.tanzu.vmware.com/addon
```

Comment:
Theorizing that this might be the wrong type, given I see:

```bash
kubectl get secret -n tkg-system | grep pinn
tkg-mgmt-vc-customized-pinn-pinniped-addon                              tkg.tanzu.vmware.com/addon            1      8m41s
tkg-mgmt-vc-customized-pinn-pinniped-data-values                        Opaque                                1      15m
tkg-mgmt-vc-customized-pinn-pinniped-fetch-0                            kubernetes.io/dockerconfigjson        1      15m
tkg-mgmt-vc-customized-pinn-pinniped-package                            clusterbootstrap-secret               1      8m37s
```
I am speculating that the type `clusterbootstrap-secret` should be on our secret here, not `tkg.tanzu.vmware.com/addon` and that the ClusterBoostrapController may not be seeing what its looking for and then is dumping out the default generic secret instead.

Reply:
We ended up using clusterbootstrap-secret as the type.